sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClientBuilder.java

```diff
@@ -573,4 +574,18 @@ public TableClientBuilder tableName(String tableName) {
 
         return this;
     }
+
+    /**
+     * Sets a flag indicating if tenant discovery should be enabled when authenticating with the Storage Table Service.
+     * This is set to {@code false} by default.
+     *
+     * @param enabled Value indicating whether tenant discovery is enabled or not.
+     *
+     * @return The updated {@link TableClientBuilder}.
+     */
+    public TableClientBuilder enableTenantDiscovery(boolean enabled) {
```

Comment:
We generally don't accept a param for on/off settings in builder. So, this would just be `enableTenantDiscovery()` without boolean arg.

See [EventHubClientBuilder](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java#L412), [ServiceBusClientBuilder](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java#L334)